src/lantern/src/JITExecute.cpp

```
@@ -0,0 +1,185 @@
+  if (length <= 0) {
+    throw std::runtime_error("Operator not found.");
+  } else if (length == 1) {
+    const auto& op {ops.front()};
+    op->getOperation()(stack_);
+  } else {
+    LLOG("Found several matches; looking for first viable schema.\n")
+    int tries {0};
+    bool found {false};
+    for (const auto& op : ops) {
+      const auto& schema {op->schema()};
+      const auto& arguments {schema.arguments()};
+      const auto num_args {arguments.size()};
+      const auto num_given {stack_.size()};
+      if (num_given != num_args) {
+        LLOG("Schema skipped: Number of arguments given does not match expected.\n")
+        tries++;
+        continue;
+      }
+      try {
+        op->getOperation()(stack_);
+        tries++;
+        found = true;
+        break;
+      } catch (c10::Error e) {
+        const auto& schema {op->schema()};
+        const auto& arguments {schema.arguments()};
+        const auto& returns {schema.returns()};
+        std::string info{"Couldn't call operator, schema is:\n"};
+        info += "Arguments:\n";
+        for (const auto& arg : arguments) {
+          info += "  Name: " + std::string(arg.name()) + "\n";
+          info += "  Type: " + std::string(c10::typeKindToString(arg.type()->kind())) + "\n";
+        }
+        info += "Returns:\n";
+        for (const auto& ret : returns) {
+          info += "  Type: " + std::string(c10::typeKindToString(ret.type()->kind())) + "\n";
+        }
+        info += "C10 error is: " + std::string(e.what_without_backtrace());
+        info += "\nTrying next.\n";
+        LLOG(info.c_str())
+        std::cout << info; // tbd remove at some later time
+      }
+    }
+  if (found) {
+    LLOG(("Found matching schema in try: " + std::to_string(tries)).c_str())
+  } else {
+    // we return the original Stack (maybe change that logic)
+    LLOG("Tried all schemas; none matched.\n")
+    std::cout << "Tried all schemas; none matched.\n";
```

We probably want to throw an error with that information. It would be weird if no error is raised but the function didn't work.